src/databricks/labs/blueprint/paths.py

```diff
@@ -448,17 +441,296 @@ def match(self, path_pattern, *, case_sensitive=None):
+    ) -> Generator[P, None, None]:
+        pattern_parts = ("**", *self._prepare_pattern(pattern))
+        if case_sensitive is None:
+            case_sensitive = True
+        selector = _Selector.parse(pattern_parts, case_sensitive=case_sensitive)
+        yield from selector(self)
+
+    @abstractmethod
+    def as_uri(self) -> str: ...
+
+    @abstractmethod
+    def as_fuse(self) -> Path: ...
+
+    @abstractmethod
+    def exists(self, *, follow_symlinks: bool = True) -> bool: ...
+
+    @abstractmethod
+    def mkdir(self, mode: int = 0o600, parents: bool = True, exist_ok: bool = True) -> None: ...
+
+    @abstractmethod
+    def rmdir(self, recursive: bool = False) -> None: ...
+
+    @abstractmethod
+    def unlink(self, missing_ok: bool = False) -> None: ...
+
+    @abstractmethod
+    def open(
+        self,
+        mode: str = "r",
+        buffering: int = -1,
+        encoding: str | None = None,
+        errors: str | None = None,
+        newline: str | None = None,
+    ): ...
+
+    @abstractmethod
+    def is_dir(self) -> bool: ...
+
+    @abstractmethod
+    def is_file(self) -> bool: ...
+
+    @abstractmethod
+    def expanduser(self: P) -> P: ...
+
+    @abstractmethod
+    def iterdir(self: P) -> Generator[P, None, None]: ...
+
+
+class DBFSPath(_DatabricksPath):
+    """Experimental implementation of pathlib.Path for DBFS paths."""
+
+    __slots__ = (
+        # The cached _file_info value for the instance.
+        "_cached_file_info",
+    )
+    _cached_file_info: FileInfo
+
+    @classmethod
+    def _from_file_info(cls, ws: WorkspaceClient, file_info: FileInfo) -> DBFSPath:
+        """Special (internal-only) constructor that creates an instance based on FileInfo."""
+        if not file_info.path:
+            msg = f"Cannot initialise without file path: {file_info}"
+            raise ValueError(msg)
+        path = cls(ws, file_info.path)
+        path._cached_file_info = file_info
+        return path
+
+    as_uri = _na("as_uri")
+
+    def as_fuse(self) -> Path:
         """Return FUSE-mounted path in Databricks Runtime."""
         if "DATABRICKS_RUNTIME_VERSION" not in os.environ:
             logger.warning("This method is only available in Databricks Runtime")
-        return Path("/Workspace", self.as_posix().lstrip("/"))
+        return Path("/dbfs", self.as_posix().lstrip("/"))
 
-    def home(self):  # pylint: disable=arguments-differ
-        """Return the user's home directory. Adapted from pathlib.Path"""
-        return WorkspacePath(self._ws, "~").expanduser()
+    def exists(self, *, follow_symlinks: bool = True) -> bool:
+        """Return True if the path points to an existing file, directory, or notebook"""
+        if not follow_symlinks:
+            raise NotImplementedError("follow_symlinks=False is not supported for DBFS")
+        try:
+            self._ws.dbfs.get_status(self.as_posix())
+            return True
+        except NotFound:
+            return False
```

Comment:
```suggestion
        return self._ws.dbfs.exists(self.as_posix())
```
i was writing that method for it to be used :) 

https://databricks-sdk-py.readthedocs.io/en/latest/workspace/files/dbfs.html#databricks.sdk.service.files.DbfsExt.exists

Reply:
I missed that, and would have used it if I'd noticed it: sorry.

That said since this comment the code has changed slightly: given that we're making an expensive call I now take the opportunity to cache the `FileInfo` result to avoid subsequent lazy lookups. What do you think about this?